src/main/java/org/broadinstitute/hellbender/utils/test/CommandLineProgramTestInterface.java

```diff
@@ -0,0 +1,94 @@
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Utility interface for CommandLine Program testing.
+ */
+public interface CommandLineProgramTestInterface {
+
+    /**
+     * Returns the name for the tested tool.
+     */
+    public String getTestedToolName();
+
+    /**
+     * For testing support. Given a name of a Main CommandLineProgram and it's arguments, builds the arguments appropriate for calling the
+     * program through Main.
+     *
+     * Default behaviour uses {@link #makeCommandLineArgs(List, String)} with the tool name provided by {@link #getTestedToolName()}.
+     *
+     * @param args List<String> of command line arguments
+     * @return String[] of command line arguments
+     */
+    default String[] makeCommandLineArgs(final List<String> args) {
+        return makeCommandLineArgs(args, getTestedToolName());
+    }
+
+    /**
+     * For testing support. Given a name of a Main CommandLineProgram and it's arguments, builds the arguments appropriate for calling the
+     * program through Main.
+     *
+     * Default behaviour generates a command line in the form "toolname args", with the verbosity parameter returned by {@link #injectDefaultVerbosity(List)}.
+     *
+     * @param args     List<String> of command line arguments
+     * @param toolname name of the tool to test
+     * @return String[] of command line arguments
+     */
+    default String[] makeCommandLineArgs(final List<String> args, final String toolname) {
+        List<String> curatedArgs = injectDefaultVerbosity(args);
+        final String[] commandLineArgs = new String[curatedArgs.size() + 1];
+        commandLineArgs[0] = toolname;
+        int i = 1;
+        for (final String arg : curatedArgs) {
+            commandLineArgs[i++] = arg;
+        }
+        return commandLineArgs;
+    }
+
+    /**
+     * Inject the verbosity parameter into the list.
+     *
+     * Default behaviour look for --verbosity argument; if not found, supply a default value that minimizes the amount of logging output.
+     */
+    default List<String> injectDefaultVerbosity(final List<String> args) {
+
+        // global toggle for BunnyLog output.
+        BunnyLog.setEnabled(false);
+
+        for (String arg : args) {
+            if (arg.equalsIgnoreCase("--" + StandardArgumentDefinitions.VERBOSITY_NAME) || arg.equalsIgnoreCase("-" + StandardArgumentDefinitions.VERBOSITY_NAME)) {
+                return args;
+            }
+        }
+        List<String> argsWithVerbosity = new ArrayList<>(args);
+        argsWithVerbosity.add("--" + StandardArgumentDefinitions.VERBOSITY_NAME);
+        argsWithVerbosity.add(Log.LogLevel.ERROR.name());
+        return argsWithVerbosity;
+    }
+
+    /**
+     * Runs the command line implemented by this test.
+     *
+     * Default behaviour uses {@link Main} with the command line arguments created by {@link #makeCommandLineArgs(List)}.
+     */
+    default Object runCommandLine(final List<String> args) {
```

Comment:
It can be abstract, but if at the end you change the Main to be used by whatever other program, will be nice to have a default one for the API user that don't care about another main.